### PR TITLE
refactor: refactor Path tracking to compose

### PR DESCRIPTION
**Pull request by Aditya-gupta99**
Jira [MIFOSAC-178](https://mifosforge.jira.com/browse/MIFOSAC-178)

<center>
    <table>
        <tr>
            <td align="center" valign="center">
<br><center>XML Screen</center>
<video src="https://github.com/openMF/android-client/assets/94394661/40cec9e8-16fb-4bf2-90b5-87942f160a40"> 
            </td>
            <td align="center" valign="center">
<br><center>Jetpack Compose</center>
<video src="https://github.com/openMF/android-client/assets/94394661/604e7aa7-7c06-47bc-9886-2078aa2893c0">                 
            </td>
        </tr>
    </table>
</center>

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

[MIFOSAC-178]: https://mifosforge.jira.com/browse/MIFOSAC-178?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: feature/path-tracking/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+plugins {
+    alias(libs.plugins.mifos.android.feature)
+    alias(libs.plugins.mifos.android.library.compose)
+    alias(libs.plugins.mifos.android.library.jacoco)
+}
+
+android {
+    namespace = "com.mifos.feature.path_tracking"
```
Comment (therajanmaurya) [inline feature/path-tracking/build.gradle.kts:8]: package name should not contains _, rename package name  "com.mifos.feature.path.tracking"

File: feature/path-tracking/src/main/java/com/mifos/feature/path_tracking/PathTrackingScreen.kt [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+
+    val context = LocalContext.current
+    val viewModel: PathTrackingViewModel = hiltViewModel()
+    val state by viewModel.pathTrackingUiState.collectAsStateWithLifecycle()
+    val refreshState by viewModel.isRefreshing.collectAsStateWithLifecycle()
+    val userStatus by viewModel.userStatus.collectAsStateWithLifecycle()
+
+    DisposableEffect(Unit) {
+
+        val notificationReceiver = object : BroadcastReceiver() {
```
Comment (therajanmaurya) [inline feature/path-tracking/src/main/java/com/mifos/feature/path_tracking/PathTrackingScreen.kt:91]: Why are you using BroadcastReceiver

File: feature/path-tracking/src/main/java/com/mifos/feature/path_tracking/PathTrackingScreen.kt [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+    userStatus: Boolean,
+    updateUserStatus: (Boolean) -> Unit
+) {
+
+    val snackbarHostState = remember { SnackbarHostState() }
+    val pullRefreshState = rememberPullRefreshState(
+        refreshing = refreshState,
+        onRefresh = onRefresh
+    )
+    val permissionState = rememberMultiplePermissionsState(
```
Comment (therajanmaurya) [inline feature/path-tracking/src/main/java/com/mifos/feature/path_tracking/PathTrackingScreen.kt:148]: Connect with Aditya (Mister K) and implement generic permission.

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

File: feature/path-tracking/build.gradle.kts (hunk 1 above)
Comment (Aditya-gupta99) [reply feature/path-tracking/build.gradle.kts]: Done

File: feature/path-tracking/src/main/java/com/mifos/feature/path_tracking/PathTrackingScreen.kt (hunk 3 above)
Comment (Aditya-gupta99) [reply feature/path-tracking/src/main/java/com/mifos/feature/path_tracking/PathTrackingScreen.kt]: Done

File: feature/path-tracking/src/main/java/com/mifos/feature/path_tracking/PathTrackingScreen.kt (hunk 2 above)
Comment (Aditya-gupta99) [reply feature/path-tracking/src/main/java/com/mifos/feature/path_tracking/PathTrackingScreen.kt:91]: This was previously implemented. it receives datas from path tracking service, but right now path service is not working, so for now, I simply added this so we don't forget this implementation. once path tracking service is fixed, if this receiver is no longer useful, I'll remove it.

Comment (therajanmaurya) [reply feature/path-tracking/src/main/java/com/mifos/feature/path_tracking/PathTrackingScreen.kt:91]: Create a ticket to migrate service in workmanager

**Pull request merged**